### PR TITLE
Datasets list

### PR DESCRIPTION
**Pull request by katieshakman**
Cleaned up comments in the example get_datasets.py.

### CODE REVIEW

File: domino/domino.py [hunk 1]
```diff
@@ -349,6 +350,41 @@ def model_version_publish(self, model_id, file, function, environment_id,
...
+        return self._get(url)
+
+    def datasets_update_details(self, datasetId, datasetName, datasetDescription):
+        self.requires_at_least("3.6.0")
+        url = self._routes.datasets_details(datasetId, datasetName, datasetDescription)
+        request = {
+            "datasetName": datasetName,
+            "description": datasetDescription
+        }
+        print(url)
```
Comment (abhijeet2096) [inline domino/domino.py:382]: please remove print statements

File: domino/routes.py [hunk 2]
```diff
@@ -105,3 +105,17 @@ def deployment_version(self):
...
+    def datasets_list(self, project_id):
+        if project_id is None:
+            return self.host + '/dataset'
+        else: 
+            return self.host + '/dataset?projectId=' + str(project_id)
+
+    def datasets_create(self): 
+        return self.host + '/dataset'
+
+    def datasets_details(self, dataset_id, dataset_name=None, dataset_description=None):
```
Comment (abhijeet2096) [inline domino/routes.py:119]: parameter `dataset_name` and `dataset_description` are passed but not used, please remove them

File: domino/domino.py [hunk 3]
```diff
@@ -349,6 +350,41 @@ def model_version_publish(self, model_id, file, function, environment_id,
         response = requests.post(url, auth=('', self._api_key), json=request)
         return response.json()
 
+    # Datasets Functions
+    def datasets_list(self, projectId=None):
+        self.requires_at_least("3.6.0")
+        url = self._routes.datasets_list(projectId)
+        return self._get(url)
+
+    def datasets_create(self, projectId, datasetName, datasetDescription):
```
Comment (abhijeet2096) [inline domino/domino.py:359]: parameter `projectId` is passed but not used, please remove it

File: domino/domino.py [hunk 4]
```diff
@@ -349,6 +350,41 @@ def model_version_publish(self, model_id, file, function, environment_id,
...
+
+    def datasets_update_details(self, datasetId, datasetName, datasetDescription):
+        self.requires_at_least("3.6.0")
+        url = self._routes.datasets_details(datasetId, datasetName, datasetDescription)
+        request = {
+            "datasetName": datasetName,
+            "description": datasetDescription
+        }
+        print(url)
+        response = requests.patch(url, data=request, auth=('', self._api_key))
```
Comment (abhijeet2096) [inline domino/domino.py:383]: I tried this function `datasets_update_details`, it didn't worked for me. It throwed this error 
```
>>> qs.datasets_update_details("5e7db98a9870cb46291f4305", "new_winds", "new description")
https://maksim160405.deploys-kappa.domino.tech/dataset/5e7db98a9870cb46291f4305
<PreparedRequest [PATCH]>
Unsupported Media Type
<Response [415]>
 ```

Changing `data` parameter in `requests.patch` to `json` works.
```
response = requests.patch(url, json=request, auth=('', self._api_key))
```
 Can you have another look on it. Thanks

File: examples/get_datasets.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+print('Datasets:'+str(raw_datasets))
+
+# Example using the current project's projectId to narrow down the datasets results to a specific project.  
+raw_datasets_for_project = domino.datasets_list(domino._project_id)  
+print('Datasets for current project:'+str(raw_datasets_for_project))
+
+# Get the details of a dataset, if one exists for the current project
+dataset_id = raw_datasets_for_project[0]['datasetId'] 
+raw_datasets_details_for_project = domino.datasets_details(dataset_id)
+print('Dataset details for current project and specified dataset: ', raw_datasets_details_for_project)
```
Comment (abhijeet2096) [inline examples/get_datasets.py:18]: Please add a new line at the end

**Comment by katieshakman** [comment]
Thanks this is on my radar but haven't been able to get to it yet due to customer items.

**Comment by ddl-olsonJD** [comment]
pr imported
- https://github.com/dominodatalab/python-domino/pull/136

**Pull request closed without merging**